Places/Scenes/CreatePlace/CreatePlaceInteractor.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  CreatePlaceInteractor.swift
+//  Places
+//
+//  Created by Andressa Valengo on 24/01/25.
+//  Copyright (c) 2025 ___ORGANIZATIONNAME___. All rights reserved.
+//
+
+import UIKit
+
+protocol CreatePlaceBusinessLogic {
+  var placeToEdit: Place? { get }
+  func isPlaceNameValid(name: String) -> Bool
+  func createPlace(request: CreatePlace.CreatePlace.Request)
+  func showPlaceToEdit(request: CreatePlace.EditPlace.Request)
+  func updatePlace(request: CreatePlace.UpdatePlace.Request)
+}
+
+protocol CreatePlaceDataStore {
+  var placeToEdit: Place? { get set }
+}
+
+class CreatePlaceInteractor: CreatePlaceBusinessLogic, CreatePlaceDataStore {
+  
+  var presenter: CreatePlacePresentationLogic?
+  
+  var placesWorker = PlacesWorker(placesStore: PlacesMemStore())
+  var placeToEdit: Place?
+  
+  // MARK: - Create place
+  
+  func createPlace(request: CreatePlace.CreatePlace.Request) {
+    let placeToCreate = buildPlaceFromPlaceInputFields(request.placeInputFields)
+    
+    placesWorker.createPlace(placeToCreate: placeToCreate) { (place: Place?) in
+      self.placeToEdit = place
+      let response = CreatePlace.CreatePlace.Response(place: place)
+      self.presenter?.presentCreatedPlace(response: response)
+    }
+  }
+  
+  // MARK: - Edit place
+  
+  func showPlaceToEdit(request: CreatePlace.EditPlace.Request) {
+    if let placeToEdit = placeToEdit {
+      let response = CreatePlace.EditPlace.Response(place: placeToEdit)
+      presenter?.presentPlaceToEdit(response: response)
+    }
+  }
+  
+  // MARK: - Update place
+  
+  func updatePlace(request: CreatePlace.UpdatePlace.Request) {
+    let placeToUpdate = buildPlaceFromPlaceInputFields(request.placeInputFields)
+    
+    placesWorker.updatePlace(placeToUpdate: placeToUpdate) { (place) in
```

Comment:
```suggestion
    placesWorker.updatePlace(placeToUpdate: placeToUpdate) { [weak self] (place) in
```
Weak self aqui para evitar memory leak